File: practise/leetcode/DailyQuestion/202410/3200. 三角形的最大高度.ts
function maxHeightOfTriangle(red: number, blue: number): number {
    return Math.max(solve(red,blue),solve(blue,red));
};

function solve(x: number, y: number): number{
    const a = 2 * Math.floor(Math.sqrt(x)) - 1;
    const b = 2 * Math.floor((-1+Math.sqrt(1 + 4 * y)) / 2);
    return Math.min(a,b) + 1;
}

File: practise/leetcode/DailyQuestion/202410/3200. 三角形的最大高度.cs
public class Solution {
    public int MaxHeightOfTriangle(int red, int blue) {
        return Math.Max(Solve(red,blue),Solve(blue,red));
    }

    public int Solve(int x,int y){
        int a = 2 * (int)(Math.Sqrt(x)) - 1;
        int b = 2 * (int)((-1+Math.Sqrt(1 + 4 * y)) / 2);
        return Math.Min(a,b) + 1;
    }
}